Sanitize `port` in `addpeeraddress()`

amadeuszpawlik opened the pull request:
In connection to #22087, it has been [pointed out](https://github.com/bitcoin/bitcoin/pull/22087#pullrequestreview-674786285) that `addpeeraddress` needs to get its port-value sanitized.

prayank23 commented:
> Or should I rather validate port directly in addpeeraddress() and just ignore the -0 case?

I think it can be ignored. Tried using `-0` in few other arguments for other RPC and it is considered as `0`. I could not find a way to read `peers.dat` however this observation is based on usage of negative zero in other parameters. 

Based on reading few posts like https://softwareengineering.stackexchange.com/questions/280648/why-is-negative-zero-important it could be an issue in some cases but not here I guess. I have no opinion on this because I couldn't do anything interesting with `-0` in `addpeeraddress`

amadeuszpawlik commented:
@prayank23 good find with `addnode` (although in that particular command the port is specified in an address string, so this wouldn't fix it). It looks like `-0` is not the only case that `addnode` accepts.
The `SplitHostPort` I propose in #22087 would definitely be nice here for port validation.
I'll go trough other RPCs and add validation for them too.

My question still stands, is modifying `univalue` a valid approach for commands that currently use `int` for port value, or do I ignore the `-0` case and simply validate directly in `net.cpp`?

MarcoFalke commented on src/rpc/net.cpp:
```diff
@@ -943,7 +943,7 @@ static RPCHelpMan addpeeraddress()
 
     const std::string& addr_string{request.params[0].get_str()};
-    const uint16_t port{static_cast<uint16_t>(request.params[1].get_int())};
+    const uint16_t& port{request.params[1].get_uint16()};
```
```suggestion
    const auto port{request.params[1].getInt<uint16_t>()};
```

on master this is now trivial to implement in a one-line diff

amadeuszpawlik committed:
Sanitize port in `addpeeraddress()`

- Ensures port sanitization in `addpeeraddress()`
- Adds test to check for invalid port values

fanquake reviewed (approved):
ACK ada8358ef54aaa04c9182afe115d8046c801bdde
